src/Mod/RuleItem.cpp

```diff
@@ -2616,6 +2616,20 @@ std::string debugDisplayScript(const RuleItem* ri)
 	}
 }
 
+void getHaveCategory(const RuleItem* ri, int& val, const std::string& cat)
+{
+	if (ri)
+	{
+		auto it = std::find(ri->getCategories().begin(), ri->getCategories().end(), cat);
+		if (it != ri->getCategories().end())
+		{
+			val = 1;
+		}
+		return;
```

Comment:
`return` return uninitialized `val`

Reply:
in case where you do not find any thing

Reply:
I agree, fix added

Reply:
Quite a few lines of code for such a simple query... why not just set val = 0 at the beginning once... and remove the duplicity. Also 'return' would not be needed?

@Yankes I see 'return' is used also in a few other script functions, is there a specific reason for that?

Lastly, please squash and force-push the commit. 1 feature = 1 commit.

Reply:
I would also say that
```
if (it != ri->getCategories().end())
{
	val = 1;
	return;
}
val = 0;
```
would also work fine in all cases without any else condition. As for English for naming it's up to you guys, I was not sure here so used syntax from our PM conversation =) Once we agree on everything I can redo it with a single commit (BTW, I am a newbie to git and I know only how to squash commits when merging a pull request, but I don't know how to merge this 2 commits in PR without redoing all the branch.

Reply:
> 
> 
> Quite a few lines of code for such a simple query... why not just set val = 0 at the beginning once... and remove the duplicity. Also 'return' would not be needed?
> 
> @Yankes I see 'return' is used also in a few other script functions, is there a specific reason for that?
> 

Function retuning `void` is more canonical version, as is map directly to operation arguments in scripts.
But for simplify binding existing function I have multiple overload that allow binding multiple different functions types and convert them to one type under hood.
(there is third case too that but its for internal script use)

For naming, I fine with any reasonable name for this function.